File: Litvinov group/semester 1/06.HW/6.3/stack.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

struct StackElement {
    string str;
    StackElement *next;
};

struct Stack {
    StackElement *head;
};

Stack* createStack() {
    return new Stack{};
}

void push(Stack *stack, const string& str) {
    StackElement *newElement = new StackElement{str, stack->head};
    stack->head = newElement;
}

string pop(Stack *stack) {
    StackElement *temp = stack->head;
    stack->head = temp->next;
    string answer = temp->str;
    delete temp;
    return answer;
}

void deleteStack(Stack* stack) {
    StackElement* temp = stack->head;
    while (temp != nullptr) {
        StackElement* del = temp;
        temp = temp->next;
        delete del;
    }
    delete stack;
}

bool isEmpty(Stack* stack) {
    return stack->head == nullptr;
}

File: Litvinov group/semester 2/2.HW/2.1/2.1/Stack.cs
using System;
using static System.Console;

namespace Stack
{
    class Stack
    {
        private class Element
        {
            public Element(string str, Element next)
            {
                this.next = next;
                this.str = str;
            }

            public Element next;
            public string str;
        }

        public Stack()
        {
        }

        public void Push(string meaning)
        {
            top = new Element(meaning, top);
            ++size;
        }

        public string Pop()
        {
            if (top == null)
            {
                WriteLine("Stack`s empty");
                return "";
            }

            Element element = top;
            top = top.next;
            --size;

            return element.str;
        }

        public bool IsEmpty() => top == null;

        public int Size() => size;

        public void Clean()
        {
            size = 0;
            top = null;
        }

        private Element top;
        private int size;
    }
}
